openvdb/openvdb/tree/InternalNode.h

```diff
@@ -2272,6 +2358,161 @@ InternalNode<ChildT, Log2Dim>::getLastValue() const
+}
+
+template<typename ChildT, Index Log2Dim>
+inline const ChildT*
+InternalNode<ChildT, Log2Dim>::getChildUnsafe(Index n) const
+{
+    return this->getConstChildUnsafe(n);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setActiveStateUnsafe(Index n, bool on)
+{
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mValueMask.set(n, on);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setValueOnlyUnsafe(Index n, const ValueType& value)
+{
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mNodes[n].setValue(value);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setValueOnUnsafe(Index n)
+{
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mValueMask.setOn(n);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setValueOnUnsafe(Index n, const ValueType& value)
+{
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mNodes[n].setValue(value);
+    mValueMask.setOn(n);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setValueOffUnsafe(Index n)
+{
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mValueMask.setOff(n);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setValueOffUnsafe(Index n, const ValueType& value)
+{
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mNodes[n].setValue(value);
+    mValueMask.setOff(n);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::setChildUnsafe(Index n, ChildNodeType* child)
+{
+    // replace tile with child
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOff(n));
+    mNodes[n].setChild(child);
+    mChildMask.setOn(n);
+    mValueMask.setOff(n);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::resetChildUnsafe(Index n, ChildNodeType* child)
+{
+    // replace child with child
+    OPENVDB_ASSERT(child);
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOn(n));
+    delete mNodes[n].getChild();
+    mNodes[n].setChild(child);
+}
+
+template<typename ChildT, Index Log2Dim>
+inline ChildT*
+InternalNode<ChildT, Log2Dim>::stealChildUnsafe(Index n, const ValueType& value, bool active)
+{
+    // replace child with tile (and return child)
+    OPENVDB_ASSERT(n < NUM_VALUES);
+    OPENVDB_ASSERT(mChildMask.isOn(n));
+    auto* child = mNodes[n].getChild();
+    mChildMask.setOff(n);
+    mValueMask.set(n, active);
+    mNodes[n].setValue(value);
+    return child;
+}
+
+template<typename ChildT, Index Log2Dim>
+inline void
+InternalNode<ChildT, Log2Dim>::deleteChildUnsafe(Index n, const ValueType& value, bool active)
```

Comment:
in the interest of minimizing code duplication and making it more intuitive this method should be implemented in terms of the previous on:

```
template<typename ChildT, Index Log2Dim>
inline void
InternalNode<ChildT, Log2Dim>::deleteChildUnsafe(Index n, const ValueType& value, bool active)
{
   delete this->stealChildUnsafe(n, value, active);
}
```